src/lib/pickles/step_main.ml

```diff
@@ -225,6 +225,15 @@ let step_main :
         and prevs =
           exists (Prev_typ.f prev_typs) ~request:(fun () ->
               Req.Proof_with_datas )
+        and unfinalized_proofs =
```

Comment:
I guess later we can have the unfinalized proofs get computed right?

Reply:
Right, the next PR in the chain (#11014) delays this calculation until after we've run the circuit's `main` code.